### PR TITLE
support for CycloneV SoC Development Kit

### PR DESCRIPTION
**Pull request by mevalver**
Signed-off-by: mevalver <esteban.valverde.vega@intel.com>

	cylonev_socdk: add board support

	this includes files under ./boards/arm/cyclonev_socdk
	this includes files under ./soc/arm/intel_socfpga_std
	added a new runner for the DevKit scripts/west_commands/runners/intel_cyclonev.py

	modified: runner selection file scripts/west_commands/runners/__init__.py

	example: lcd_display: created a lcd_display example for Cyclone V SoC development kit samples/drivers/lcd_cyclonev_socdk

	dts: dts files for the cyclone v soc development kit board dts/arm/intel_socfpga_std

	i2c: modified driver i2c_dw.c to use with this board

	serial: driver: modified driver uart_ns16550.c to work with extended fifo

	serial: driver: kconfig: adding extended fifo configuration 128 bytes

	usb: driver: usb_dc_dw.c adding lines to support GIC_V1

### CODE REVIEW

File: samples/drivers/lcd_cyclonev_socdk/src/main.c [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2022, Intel Corporation
+ * Description:
+ * Example to use LCD Display in Cyclone V SoC FPGA devkit
+ */
+
+#include <zephyr.h>
```
Comment (teburd) [inline samples/drivers/lcd_cyclonev_socdk/src/main.c:8]: zephyr.h and kernel.h below are effectively the same thing, the can both be replaced with <zephyr/kernel.h>

File: boards/arm/cyclonev_socdk/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+.. code-block:: console
+
+   $ quartus_pgm -m jtag -o "p;path/to/soc_system.sof"
+
+This system is composed by the HPS, ARM Cortex-A9. In this example the UART, timer,
+USB, I2C, DDR memory are exposed. Please double check the peripheral you intend to 
+use have its corresponding driver support. 
+You can find more information of the Cyclone® V SoC Devkit GSRD in RocketBoards
+or consult the "Cyclone® V HArd Processor System Technical Reference Manual"
```
Comment (teburd) [inline boards/arm/cyclonev_socdk/doc/index.rst:142]: HArd -> Hard

**Comment by mevalver** [comment]
closing to make a new one cleaner

**Comment by MaureenHelm** [comment]
> closing to make a new one cleaner

In the future, please update your pull requests instead. See https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow

**Pull request closed without merging**